users/twschum/xtonhasvim.c

```diff
@@ -0,0 +1,630 @@
+ /* Copyright 2015-2017 Christon DeWan
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "xtonhasvim.h"
+
+/************************************
+ * helper foo
+ ************************************/
+
+#define PRESS(kc) register_code(kc)
+#define RELEASE(kc) unregister_code(kc)
+
+static inline void TAP(uint16_t keycode) {
+    PRESS(keycode);
+    RELEASE(keycode);
+}
+
+static inline void CMD(uint16_t keycode) {
+  PRESS(KC_LGUI);
+    TAP(keycode);
+  RELEASE(KC_LGUI);
+}
+
+static inline void CTRL(uint16_t keycode) {
+  PRESS(KC_LCTRL);
+    TAP(keycode);
+  RELEASE(KC_LCTRL);
+}
+
+static inline void SHIFT(uint16_t keycode) {
+  PRESS(KC_LSHIFT);
+    TAP(keycode);
+  RELEASE(KC_LSHIFT);
+}
+
+static inline void ALT(uint16_t keycode) {
+  PRESS(KC_LALT);
+    TAP(keycode);
+  RELEASE(KC_LALT);
+}
```

Comment:
Honestly, none of this is needed.

`tap_code` allows for basic keycodes to be used, and `tap_code16` allofr for modded keycodes, (such as `tap_code16(LSFT(KC_I))`, or `tap_code16(LALT(KC_TAB))`